pandas/core/indexing.py

```diff
@@ -2207,10 +2208,12 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
                 "the indexed object do not match)."
             )
         result = result.astype(bool)._values
-    else:
+    elif is_object_dtype(key):
         # key might be sparse / object-dtype bool, check_array_indexer needs bool array
```

Comment:
The comment here seems to indicate that this was also done for sparse data (and not only object dtype data). But no tests are failing? This might need some checking if the comment was outdated (I think it was only recently added)

Reply:
Can you check this comment? (about the sparse from the code comment)